lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -753,9 +753,9 @@ def recurring_parse_element(response, node)
         end
       end
 
-      def commit(order, message_type, trace_number = nil)
+      def commit(order, message_type, retry_logic = nil, trace_number = nil)
         headers = POST_HEADERS.merge('Content-length' => order.size.to_s)
-        if @options[:retry_logic] && trace_number
+        if @options[:retry_logic] || retry_logic && trace_number
```

Comment:
it's late, but i _think_ that `&&` takes precedence over `||`. So
`if @options[:retry_logic] || retry_logic && trace_number` 
is not the same as 
`if (@options[:retry_logic] || retry_logic) && trace_number` (which would preserve the current behavior):

```
irb(main):016:0> true || false && nil
=> true
irb(main):017:0> (true || false) && nil
=> nil
```

So if we still only want to set the headers if there's a trace number in either case, we will want some parens here. (It looks like we don't currently have a test case that checks that we don't send the headers if there's no trace number.)

Reply:
Oh good catch, I meant to wrap them and forgot! And yeah I'll add in a test for "if we don't send retry logic than ignore trace number" and also "if there's no trace number than ignore retry logic check"